src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -165,6 +165,12 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, ArgumentTypes);
             }
 
+            // Array construction is a special case.
+            if (DeclaringType is not null && DeclaringType.IsArray)
+            {
+                return InvokeArrayCtorWorker(arguments);
```

Comment:
Before this change, the argument array reflected the coerced argument values. It does not seem to be the case anymore now that we are skipping the copy-out below.

E.g. this prints `System.Int32` today:
```csharp
using System.Globalization;
using System.Reflection;

var ctor = typeof(int[,]).GetConstructor(new[] { typeof(int), typeof(int) });
object[] arguments = new object[] { "1", "2" };
var arr = ctor.Invoke(BindingFlags.Default, new MyBinder(), arguments, null);
Console.WriteLine(arguments[0].GetType());

class MyBinder : Binder
{
    public override FieldInfo BindToField(BindingFlags bindingAttr, FieldInfo[] match, object value, CultureInfo? culture)
        => throw new NotImplementedException();

    public override MethodBase BindToMethod(BindingFlags bindingAttr, MethodBase[] match, ref object?[] args, ParameterModifier[]? modifiers, CultureInfo? culture, string[]? names, out object? state)
        => throw new NotImplementedException();

    public override object ChangeType(object value, Type type, CultureInfo? culture)
        => int.Parse((string)value);

    public override void ReorderArgumentArray(ref object?[] args, object state)
        => throw new NotImplementedException();

    public override MethodBase? SelectMethod(BindingFlags bindingAttr, MethodBase[] match, Type[] types, ParameterModifier[]? modifiers)
        => throw new NotImplementedException();

    public override PropertyInfo? SelectProperty(BindingFlags bindingAttr, PropertyInfo[] match, Type? returnType, Type[]? indexes, ParameterModifier[]? modifiers)
        => throw new NotImplementedException();
}
```

Reply:
Bah. Wonderful consequence of this behavior. Fixed and added test.